homeassistant/components/media_player/sonos.py

```diff
@@ -66,6 +66,26 @@ def setup_platform(hass, config, add_devices, discovery_info=None):
     return True
 
 
+def only_if_coordinator(func):
+    """
+    If used as decorator, avoid calling the decorated method if
+    player is not a coordinator.
+    If not, a grouped speaker (not in coordinator role)
+    will throw soco.exceptions.SoCoSlaveException
+    """
+
+    def wrapper(*args):
```

Comment:
It was not needed for the use case, but I guess it might be handy in the future. Adding